L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -0,0 +1,772 @@
+
+  /// ///////////////// ///
+  /// MANDATORY METHODS ///
+  virtual void beginRun(const edm::Run& run, const edm::EventSetup& iSetup);
+  virtual void endRun(const edm::Run& run, const edm::EventSetup& iSetup);
+  virtual void produce(edm::Event& iEvent, const edm::EventSetup& iSetup);
+  virtual void endJob();
+};
+
+//////////////
+// CONSTRUCTOR
+L1FPGATrackProducer::L1FPGATrackProducer(edm::ParameterSet const& iConfig)
+    : config(iConfig),
+      readMoreMcTruth_(iConfig.getParameter<bool>("readMoreMcTruth")),
+      MCTruthClusterInputTag(readMoreMcTruth_ ? config.getParameter<edm::InputTag>("MCTruthClusterInputTag")
+                                              : edm::InputTag()),
+      MCTruthStubInputTag(readMoreMcTruth_ ? config.getParameter<edm::InputTag>("MCTruthStubInputTag")
+                                           : edm::InputTag()),
+      TrackingParticleInputTag(readMoreMcTruth_ ? iConfig.getParameter<edm::InputTag>("TrackingParticleInputTag")
+                                                : edm::InputTag()),
+      TrackingVertexInputTag(readMoreMcTruth_ ? iConfig.getParameter<edm::InputTag>("TrackingVertexInputTag")
+                                              : edm::InputTag()),
+      ttStubSrc_(config.getParameter<edm::InputTag>("TTStubSource")),
+      bsSrc_(config.getParameter<edm::InputTag>("BeamSpotSource")),
+
+      ttStubToken_(consumes<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_> > >(ttStubSrc_)),
+      bsToken_(consumes<reco::BeamSpot>(bsSrc_)) {
+  if (readMoreMcTruth_) {
+    ttClusterMCTruthToken_ = consumes<TTClusterAssociationMap<Ref_Phase2TrackerDigi_> >(MCTruthClusterInputTag);
+    ttStubMCTruthToken_ = consumes<TTStubAssociationMap<Ref_Phase2TrackerDigi_> >(MCTruthStubInputTag);
+    TrackingParticleToken_ = consumes<std::vector<TrackingParticle> >(TrackingParticleInputTag);
+    TrackingVertexToken_ = consumes<std::vector<TrackingVertex> >(TrackingVertexInputTag);
+  }
+
+  produces<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >("Level1TTTracks").setBranchAlias("Level1TTTracks");
+
+  phiWindowSF_ = iConfig.getUntrackedParameter<double>("phiWindowSF", 1.0);
+
+  asciiEventOutName_ = iConfig.getUntrackedParameter<string>("asciiFileName", "");
+
+  fitPatternFile = iConfig.getParameter<edm::FileInPath>("fitPatternFile");
+  processingModulesFile = iConfig.getParameter<edm::FileInPath>("processingModulesFile");
+  memoryModulesFile = iConfig.getParameter<edm::FileInPath>("memoryModulesFile");
+  wiresFile = iConfig.getParameter<edm::FileInPath>("wiresFile");
+
+  DTCLinkFile = iConfig.getParameter<edm::FileInPath>("DTCLinkFile");
+  moduleCablingFile = iConfig.getParameter<edm::FileInPath>("moduleCablingFile");
+
+  DTCLinkLayerDiskFile = iConfig.getParameter<edm::FileInPath>("DTCLinkLayerDiskFile");
+
+  extended_ = iConfig.getUntrackedParameter<bool>("Extended", false);
+  nHelixPar_ = iConfig.getUntrackedParameter<unsigned int>("Hnpar", 4);
```

Comment:
if these parameters affect the results of the algorithm, they should not be `untracked`

Reply:
these are no longer 'untracked' as they indeed define the algorithm.